tests/generators/test_binomial_tree.py

```diff
@@ -0,0 +1,53 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestBinomailTreeGraph(unittest.TestCase):
+
+    def test_binomial_tree_graph(self):
+        for n in range(10):
+            graph = retworkx.generators.binomial_tree_graph(n)
+            self.assertEqual(len(graph), 2**n)
+            self.assertEqual(len(graph.edges()), 2**n - 1)
+
+    def test_binomial_tree_graph_weights(self):
```

Comment:
Can you add a negative test case where `len(weights) != 2^order` ?

Reply:
Sure

Reply:
I added `test_binomial_tree_graph_weight_less_nodes` for the case when `weights < 2^order` and `test_binomial_tree_graph_weights_greater_nodes` when `weights > 2^order`